crates/rpc/rpc-builder/src/eth.rs

```diff
@@ -112,3 +117,117 @@ impl EthConfig {
         self
     }
 }
+
+/// Context for building the `eth` namespace server.
+#[derive(Debug)]
+pub struct EthServerBuilderCtx<'a, Provider, Pool, EvmConfig, Network, Tasks, Events> {
+    /// Database handle.
+    pub provider: Provider,
+    /// Mempool handle.
+    pub pool: Pool,
+    /// Network handle.
+    pub network: Network,
+    /// EVM configuration.
+    pub evm_config: EvmConfig,
+    /// RPC config for `eth` namespace.
+    pub config: &'a EthConfig,
+    /// Runtime handle.
+    pub executor: Box<Tasks>,
+    /// Events handle.
+    pub events: Events,
+    /// RPC cache handle.
+    pub cache: EthStateCache,
+    /// Transaction forwarder used by `eth_sendRawTransaction` method.
+    pub raw_transaction_forwarder: Option<Arc<dyn RawTransactionForwarder>>,
+}
+
+impl<'a, Provider, Pool, EvmConfig, Network, Tasks, Events>
+    EthServerBuilderCtx<'a, Provider, Pool, EvmConfig, Network, Tasks, Events>
+{
+    /// Creates a new context for building the `eth` namespace server.
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        provider: Provider,
+        pool: Pool,
+        network: Network,
+        evm_config: EvmConfig,
+        config: &'a EthConfig,
+        executor: Box<Tasks>,
+        events: Events,
+        cache: EthStateCache,
+        raw_transaction_forwarder: Option<Arc<dyn RawTransactionForwarder>>,
+    ) -> Self {
+        Self {
+            provider,
+            pool,
+            network,
+            evm_config,
+            config,
+            executor,
+            events,
+            cache,
+            raw_transaction_forwarder,
+        }
+    }
+}
+
+/// Builds RPC server for `eth` namespace.
+pub trait EthServerBuilder<Provider, Pool, EvmConfig, Network, Tasks, Events>:
+    Clone + Copy
+{
+    /// `eth` namespace RPC server type.
+    type Server: EthApiServer + UpdateRawTxForwarder + Clone + FullEthServer;
```

Comment:
directionally, this makes sense.

although, I think we should try to add another layer of abstraction here

1. we need some type that launches the server (rpc-builder)
2. this will require something that can create an EthApi

so perhaps we limit this one here just to `EthApi` (e.g. EthApiBuilder)

so we can eventually abstract the rpc entirely

Reply:
I was pondering on the name and thought `ApiBuilder` sounds like metaprogramming, but I'm fine with changing it

Reply:
https://en.wikipedia.org/wiki/Metaprogramming